flow/record/base.py

```diff
@@ -756,6 +756,30 @@ def fieldtype(clspath):
     return t
 
 
+@functools.lru_cache(maxsize=4069)
+def merge_record_descriptors(descriptors, replace=False, name=None):
+    """Create a newly merged RecordDescriptor from a list of RecordDescriptors.
+    This function uses a cache to avoid creating the same descriptor multiple times.
+
+    Duplicate fields are ignored in `descriptors` unless `replace=True`.
+
+    Args:
+        descriptors (Tuple[RecordDescriptor]): Tuple of RecordDescriptors we use for extending/replacing.
```

Comment:
```suggestion
        descriptors (Tuple[RecordDescriptor]): Tuple of RecordDescriptors to use for extending/replacing.
```

Reply:
Also would a type of `Sequence` or `Iterable` not be more accurate?

Reply:
In this case it has to be a Tuple, otherwise it's not hashable for lru_cache.